src/main/webapp/js/captions.js

```diff
@@ -100,34 +105,94 @@ function execute(url) {
 
     // launch yt captions request (needed for gensim in no-db model)
     ytCaptions = "";
-    getTrackId(videoId)
-        .then(trackId => getYTCaptions(trackId))
-        .then(captions => parseCaptionsIntoJson(captions))
-        .then(parsed_captions => {
-            ytCaptions = parsed_captions;
-            documents = createDocuments(parsed_captions);
+    fullCaptions = "";
+    hasCaptions = false;
+    fetch('/fullcaption?id=' + videoId, {
+            method: 'GET',
+        }).then((response) => response.json()).then(json => {
+            if (Array.isArray(json) && json.length > 0) {
+                fullCaptions = getFullCaptionsText(json);
+                json.forEach((item, index) => {
+                    json[index].text = item.text.replace(/\[.*?\]/g, '');
+                });
+                ytCaptions = JSON.stringify({'captions': json});
+                hasCaptions = true;
+                documents = createDocuments(ytCaptions);
+                console.log("Got ytCaptions from DB")
+                callGensim(videoId, ytCaptions);
+            } else {
+                console.log("Got ytCaptions from YT API")
+                getTrackId(videoId)
+                    .then(trackId => getYTCaptions(trackId))
+                    .then(captions => parseCaptionsIntoJson(captions))
+                    .then(parsed_captions => {
+                        ytCaptions = parsed_captions;
+                        documents = createDocuments(parsed_captions);
+                        callGensim(videoId, ytCaptions);
+                    });
+            }
         });
+}
 
-    // show loading text
-    $('#loading-text').show();
-
-    // checks to see if captions already exist in the database
-    fetch('/caption?id=' + videoId, {
-        method: 'GET',
-    }).then((response) => response.json()).then((json) => {
-        if (Object.keys(json).length > 0) {
-            // Sets the results table
-            successfulDisplay(json);
-            console.log("Fetching captions from database...");
-        } else {
-            // video id not found in db, fetching from Youtube API
-            getTrackId(videoId)
-            .then(trackId => getYTCaptions(trackId))
-            .then(captions => parseCaptionsIntoJson(captions, url))
-            .then(parsed_captions => sendJsonForm(parsed_captions))
-            .then(nlp_json => successfulDisplay(nlp_json));
-        }
+/**
+ * Calls the gensim server to set up the current video id and captions for processing
+ * @param video_id the string id of the YouTube video
+ * @param yt_captions the list of yt_captions
+ */
+function callGensim(video_id, yt_captions) {
```

Comment:
The name here is not that meaningful. Plz change it to sth like processVideoOnGensim

Reply:
addressed